src/main/java/slimeknights/mantle/util/TradeGeneric.java

```diff
@@ -0,0 +1,154 @@
+package slimeknights.mantle.util;
+
+import java.util.Random;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import net.minecraft.entity.passive.EntityVillager;
+import net.minecraft.entity.passive.EntityVillager.PriceInfo;
+import net.minecraft.init.Items;
+import net.minecraft.item.ItemStack;
+import net.minecraft.village.MerchantRecipe;
+import net.minecraft.village.MerchantRecipeList;
+
+public class TradeGeneric implements EntityVillager.ITradeList {
+  private ItemStack input, input2, output;
+  private PriceInfo inputPrice, input2Price, outputPrice;
+
+  /**
+   * Creates a new trade of two input stacks for an output stack
+   *
+   * @param input        First input itemstack
+   * @param inputPrice   Random stack size range for first input
+   * @param input2       Second input itemstack
+   * @param input2Price  Random stack size range for second input
+   * @param output       Output itemstack
+   * @param outputPrice  Random stack size range for output
+   */
+  public TradeGeneric(ItemStack input, PriceInfo inputPrice, ItemStack input2, PriceInfo input2Price,
+      ItemStack output, PriceInfo outputPrice) {
+
+    this.input = input;
+    this.inputPrice = inputPrice;
+
+    this.input2 = input2;
+    this.input2Price = input2Price;
+
+    this.output = output;
+    this.outputPrice = outputPrice;
+  }
+
+  /**
+   * Creates a new trade of an input stack for an output stack
+   *
+   * @param input        Input itemstack
+   * @param inputPrice   Random stack size range for input
+   * @param output       Output itemstack
+   * @param outputPrice  Random stack size range for output
+   */
+  public TradeGeneric(ItemStack input, PriceInfo inputPrice, ItemStack output, PriceInfo outputPrice) {
+    this(input, inputPrice, null, null, output, outputPrice);
+  }
+
+  /**
+   * Creates a new trade with one input for emeralds
+   *
+   * @param input        Input itemstack
+   * @param inputPrice   Random stack size range for input
+   * @param outputPrice  Random range for output emerald
+   */
+  public TradeGeneric(ItemStack input, PriceInfo inputPrice, PriceInfo outputPrice) {
+    this(input, inputPrice, null, null, null, outputPrice);
+  }
+
+  /**
+   * Creates a new trade of emeralds for an output stack
+   *
+   * @param inputPrice   Random stack size range for input
+   * @param output       Output itemstack
+   * @param outputPrice  Random range for output emerald
+   */
+  public TradeGeneric(PriceInfo inputPrice, ItemStack output, PriceInfo outputPrice) {
+    this(null, inputPrice, null, null, output, outputPrice);
+  }
+
+  /**
+   * Getter for the input stack
+   * @param random  random number access
+   * @return  A new instance of the input stack, or a stack containing an emerald if null
+   */
+  @Nonnull
```

Comment:
Technically it's not, just unused by default. See TradeRandom which extends this (which can easily be done by other classss well for more complicated random functions)